### ISSUE TITLE
Support for decimal.Decimal and other numerical types as an alternative to float

### ISSUE THREAD

**Issue by ezekielnewren** [pyparsing/plusminus#20]
I'd like there be an option like `parser = BaseArithmeticParser(numerical_types=[int, Decimal, complex])` where if `numerical_types=None` it would default to the value `numerical_types=[int, float, complex]` so that when I'm dealing with monetary values I can be confident that the answer IS EXACT, not really super close.

**Comment by ptmcg** [comment]
Interesting idea, I'll look into it.

**Comment by ptmcg** [comment]
Well it took me a while to get around to this, but I did put some time into it, and I think I have a working version now. Rather than have you poke around with the internal numerical_types variable, I've added an optional `use_decimal` Boolean argument to the BaseArithmeticParser class initializer - when set to True, the internal math and resulting values will be Decimal objects, not floats. Works for common math operations, but complex results and trig functions are not supported in this mode.
